utils/zerovec/src/varzerovec/borrowed.rs

```diff
@@ -328,29 +328,82 @@ where
     }
 }
 
-pub fn get_serializable_bytes<T: VarULE + ?Sized, A: custom::EncodeAsVarULE<T>>(
-    elements: &[A],
-) -> Option<Vec<u8>> {
-    // Assume each element is probably around 4 bytes long when estimating the
-    // initial size. Performance of this function does not matter *too* much since
-    // VarZeroVec mutation is not intended to be performant.
-    let mut vec = Vec::with_capacity(4 + 8 * elements.len());
-    let len_u32: u32 = elements.len().try_into().ok()?;
-    vec.extend(&len_u32.as_unaligned().0);
-    // Make space for indices
-    vec.resize(4 + 4 * elements.len(), 0);
-    let mut offset: u32 = 0;
-    for (idx, element) in elements.iter().enumerate() {
+/// Collects the bytes for a VarZeroSlice into a Vec.
+pub fn get_serializable_bytes<T, A>(elements: &[A]) -> Option<Vec<u8>>
+where
+    T: VarULE + ?Sized,
+    A: custom::EncodeAsVarULE<T>,
+{
+    let len = compute_serializable_len(elements)?;
+    debug_assert!(len >= 4);
+    let mut output = Vec::with_capacity(len as usize);
+    // Safety: All bytes will be initialized after calling write_serializable_bytes
+    unsafe {
+        output.set_len(len as usize);
+    }
+    write_serializable_bytes(elements, &mut output);
+    Some(output)
+}
+
+/// Writes the bytes for a VarZeroSlice into an output buffer.
+///
+/// Every byte in the buffer will be initialized after calling this function.
+///
+/// # Panics
+///
+/// Panics if the buffer is not exactly the correct length.
+pub fn write_serializable_bytes<T, A>(elements: &[A], output: &mut [u8])
```

Comment:
nit (nb): these two can probably be pub-crate since they're accessible via an EncodeAsVarULE impl. The first one is somewhat generally useful, at least for tests.

Reply:
The module `borrowed` is already pub(crate).